pkg/controller/identity/iampolicy/controller.go

```diff
@@ -0,0 +1,300 @@
+/*
+Copyright 2019 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package iampolicy
+
+import (
+	"context"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	awsarn "github.com/aws/aws-sdk-go-v2/aws/arn"
+	awsiam "github.com/aws/aws-sdk-go-v2/service/iam"
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	runtimev1alpha1 "github.com/crossplane/crossplane-runtime/apis/core/v1alpha1"
+	"github.com/crossplane/crossplane-runtime/pkg/event"
+	"github.com/crossplane/crossplane-runtime/pkg/logging"
+	"github.com/crossplane/crossplane-runtime/pkg/meta"
+	"github.com/crossplane/crossplane-runtime/pkg/reconciler/managed"
+	"github.com/crossplane/crossplane-runtime/pkg/resource"
+
+	"github.com/crossplane/provider-aws/apis/identity/v1alpha1"
+	awsv1alpha3 "github.com/crossplane/provider-aws/apis/v1alpha3"
+	awsclients "github.com/crossplane/provider-aws/pkg/clients"
+	"github.com/crossplane/provider-aws/pkg/clients/iam"
+)
+
+const (
+	errNotPolicyInstance = "managed resource is not an IAMPolicy custom resource"
+
+	errCreatePolicyClient = "cannot create IAM Policy client"
+	errGetProvider        = "cannot get provider"
+	errGetProviderSecret  = "cannot get provider secret"
+
+	errUnexpectedObject = "The managed resource is not a IAMPolicy resource"
+	errGet              = "failed to get IAM Policy"
+	errCreate           = "failed to create the IAM Policy"
+	errDelete           = "failed to delete the IAM Policy"
+	errUpdate           = "failed to update the IAM Policy"
+	errEmptyPolicy      = "empty IAM Policy received from IAM API"
+	errPolicyVersion    = "No version for policy received from IAM API"
+	errUpToDate         = "cannt check if policy is up to date"
+)
+
+// SetupIAMPolicy adds a controller that reconciles IAM Policy.
+func SetupIAMPolicy(mgr ctrl.Manager, l logging.Logger) error {
+	name := managed.ControllerName(v1alpha1.IAMPolicyGroupKind)
+
+	return ctrl.NewControllerManagedBy(mgr).
+		Named(name).
+		For(&v1alpha1.IAMPolicy{}).
+		Complete(managed.NewReconciler(mgr,
+			resource.ManagedKind(v1alpha1.IAMPolicyGroupVersionKind),
+			managed.WithExternalConnecter(&connector{kube: mgr.GetClient(), newClientFn: iam.NewPolicyClient}),
```

Comment:
```suggestion
			managed.WithExternalConnecter(&connector{kube: mgr.GetClient(), newClientFn: iam.NewPolicyClient}),
			managed.WithInitializers(),
```

I think we should add the above option to disable [the default behaviour of setting the external name to the managed resource's name](https://github.com/crossplane/crossplane-runtime/blob/290de73499493eae198345e75cf79e1608f25d56/pkg/reconciler/managed/api.go#L49).

I suspect this is working at the moment, but a little misleading, because:

1. When the managed resource is first reconciled we set the external name to the managed resource name.
1. During Observe we notice that the external name is not an ARN, so we report that the resource doesn't exist.
1. During Create we overwrite the external name with the created ARN.

Reply:
@negz Thanks for the catch, I have updated this.